apollo-federation/src/connectors/json_selection/lit_expr.rs

```diff
@@ -54,13 +55,94 @@ pub(crate) enum LitExpr {
     // of the path, which is never PathList::Empty, because that would mean the
     // LitExpr could stand on its own, using one of the other variants.
     LitPath(WithRange<LitExpr>, WithRange<PathList>),
+
+    // Operator chains: A op B op C ... where all operators are the same type
+    // OpChain contains the operator type and a vector of operands
+    // For example: A ?? B ?? C becomes OpChain(NullCoalescing, [A, B, C])
+    OpChain(LitOp, Vec<WithRange<LitExpr>>),
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub(crate) enum LitOp {
+    NullCoalescing, // ??
+    NoneCoalescing, // ?!
```

Comment:
Naming discussion/idea:
```suggestion
#[derive(Debug, PartialEq, Eq, Clone)]
pub(crate) enum LitOp {
    NullCoalescing, // ??
    NoneDefault, // ?!
```
As I understand it, the word "coalescing" in the [ECMAScript/JavaScript nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) syntax refers to the way `null` and `undefined` get handled the same way (coalesced). That's what we're aiming for with `??`, so `NullCoalescing` works for that operator name.

If we want to lean into the JS precedent even further, we might call `??` the `LitOp::NullishCoalescing` operator, where "nullish" values are `null` and `None`, but I don't think `NullCoalescing` is wrong or misleading.

However, with our `?!` operator, multiple input values are not being coalesced, but instead we are defaulting to another value when the first value is `None`/missing. I don't think `NoneCoalescing` quite fits for that, but I like `NoneDefault`. Thoughts?

Reply:
After discussion on slack, we are settling on "NullishCoalescing" and "NoneCoalescing".